resources/resource_utils.py

```diff
@@ -0,0 +1,60 @@
+from typing import Optional, Dict
+import hail as hl
+
+
+# Resource classes
+class _Resource:
+    """
+    Generic abstract resource class.
+    """
+    def __init__(self, path: str, source_path: Optional[str] = None, **kwargs):
+        if type(self) is _Resource:
+            raise TypeError("Can't instantiate abstract class _Resource")
+
+        self.source_path = source_path
+        self.path = path
```

Comment:
Some documentation explaining the difference between `source_path` and `path` would be nice. So far as I can tell, `source_path` is only used when the resource is imported from a non-Hail format? But the attribute names don't make that clear, `source_path` could easily be interpreted as the path from which a Table is read.

Reply:
I'm open, but I think I like keeping source_path -- there are really quite a few resources that will have this. Totally agree on the documentation and open to suggestions for other names!